src/DotNetty.Handlers/Tls/TlsHandler.Writer.cs

```diff
@@ -173,7 +178,18 @@ private void Wrap(IChannelHandlerContext context)
                         _lastContextWritePromise = promise;
                         if (buf.IsReadable())
                         {
+#if NETCOREAPP_2_0_GREATER || NETSTANDARD_2_0_GREATER
+                            var asyncWrite = WriteAsync(buf, promise);
+                            if (!asyncWrite.IsCompleted)
+                            {
+                                var asyncWriteTask = asyncWrite.AsTask();
+                                asyncWriteTask.Ignore();
+                                _lastAsyncWriteTask = asyncWriteTask;
+                            }
+                            buf = null; //prevent buf from releasing synchronously
```

Comment:
why in this case we dont want to call `buf?.Release()` first? What is the difference between this case and ` buf.ReadBytes(_sslStream, readableBytes);` in terms of releasement?

Reply:
that's because the buffer is passed into sslStream.WriteAsync method so it's not expected to be released before write task is completed.